reporting/lsrd_stats.py

```diff
@@ -1,675 +1,388 @@
-	
-	SQL = "select COUNT(*) from ordering_order where order_date::text like \'" + giveLastMonthDate() + "-%\' and orderid not like \'%@usgs.gov-%\' and ordering_order.order_source = '" + source + "'"
-	
-	cursor.execute(SQL)
-	
-	rows = cursor.fetchall()
-     
-	if len(rows) >= 1:
-	    for row in rows:
-		total = str(row[0])
-		
-		f.write(" Number of total orders placed in the month (non-USGS) for '" + source + "' interface: " + total + "\n")
-		#print " Number of total orders placed in the month (non-USGS):", total
-	else:
-	    f.write(" Number of total orders placed in the month (non-USGS) for '" + source + "' interface: 0\n")
-	    #print " Number of total orders placed in the month (non-USGS): 0"
-	    #print    
-	
-	#----------------------------------------------
-	# Total number of unique on-demand users 
-	#----------------------------------------------
-	
-	#SQL = "select COUNT(DISTINCT(substring_index(orderid,'-',1))) from ordering_order where order_date::text like \'" + giveLastMonthDate() + "-%\' and  ordering_order.order_source = '" + source + "'"
-        SQL = "select COUNT(DISTINCT(split_part(orderid,'-',1))) from ordering_order where order_date::text like \'" + giveLastMonthDate() + "-%\' and  ordering_order.order_source = '" + source + "'"
-	
-	cursor.execute(SQL)
-    
-	rows = cursor.fetchall()
-     
-	if len(rows) >= 1:
-	    for row in rows:
-		total = str(row[0])
-		
-		f.write(" Total number of unique On-Demand users for '" + source + "' interface: " + total + "\n")
-		#print " Total number of unique On-Demand users:", total
-	else:
-	    f.write(" Total number of unique On-Demand users for '" + source + "' interface: 0\n")
-	    #print " Total number of unique On-Demand users: 0"
-	    #print    
-    
-	f.write("\n")
-    
-    #------------end of massive loop-----------
-    
-    # Close db and report-file handles
-    db_conn.close()
-    f.close()
-    
-    # Send off an e-mail
-    
-    #datestamp_tag = giveLastMonthDate()
-    #email_subject = email_subject + " - " + datestamp_tag
-
-    f = open(outfile, "r")
-    reportbody = f.readlines()
-    
-    body = ""
-    
-    for line in reportbody:
-        body = body + line
-    
-    f.close()
-    
-    send_email(email_from,email_to,email_subject + " - " + giveLastMonthDate(),body)
-    
+
+    return args
+
+
+def setup_cron():
+    """
+    Setup cron job for the 1st of the month
+    for the previous month's metrics
+    """
+    chron_file = 'tmp'
+
+    cron_str = '00 06 1 * * /usr/local/bin/python {1} -p'.format(FILE_PATH)
+
+    crons = subprocess.check_output(['crontab', '-l']).split('\n')
+
+    for idx, line in enumerate(crons):
+        if __file__ in line:
+            print('Cron job already exists')
+            return
+
+    add = ['#-----------------------------',
+           '# LSRD ESPA stats',
+           '#-----------------------------',
+           cron_str]
+
+    crons.extend(add)
+
+    with open(chron_file, 'w') as f:
```

Comment:
please make one change... when writing a new cron, please backup current cron to the home directory of the user, name it backups/crontab-mmddyy-hhmmss